esp/src/src-react/util/throttle.ts

```diff
@@ -0,0 +1,12 @@
+import { debounce } from "@hpcc-js/util";
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type TypeOfClassMethod<T, M extends keyof T> = T[M] extends Function ? T[M] : never;
+
+export function singletonDebounce<T, M extends keyof T>(obj: T, method: M): TypeOfClassMethod<T, M> {
+    const __lazy__ = Symbol.for(`__lazy__${method}`);
+    if (!obj[__lazy__]) {
+        obj[__lazy__] = debounce((...args: any[]) => (obj[method] as any)(...args), 1000);
```

Comment:
Would there be any usefulness in exposing the wait time?

Reply:
I wasn't sure - my thought was that it should be configurable and at least this way its in one place.

FYI - this is an "early" debounce.   IOW it lets the first call through and all the matching ones within a 1000ms get the same result.  You might be more familiar with a "late" debounce for things like resizing a window etc.

Reply:
Ah, ok. Yes, that is what I was thinking of.